great_expectations/rule_based_profiler/config/base.py

```diff
@@ -99,12 +92,19 @@ class Meta:
     batch_request = fields.Dict(keys=fields.String(), required=False, allow_none=True)
 
 
-@dataclass(frozen=True)
 class ParameterBuilderConfig(DictDot):
-    name: str
-    class_name: str
-    module_name: Optional[str] = None
-    batch_request: Optional[Dict[str, Any]] = None
+    def __init__(
+        self,
+        name: str,
+        class_name: str,
+        module_name: Optional[str] = None,
+        batch_request: Optional[Dict[str, Any]] = None,
+        **kwargs
+    ):
+        self.name = name
+        self.class_name = class_name
+        self.module_name = module_name
+        self.batch_request = batch_request
```

Comment:
@cdkini I feel that this pattern has merit:
```
        for k, v in kwargs.items():
            setattr(self, k, v)
```

Happy to discuss.

Reply:
I'd like to understand the merits of this approach

Reply:
Added!